nautobot/extras/forms/forms.py

```diff
@@ -1499,6 +1511,100 @@ class SecretsGroupFilterForm(NautobotFilterForm):
     q = forms.CharField(required=False, label="Search")
 
 
+#
+# Static Groups
+#
+
+
+class StaticGroupForm(NautobotModelForm):
+    """Generic create/update form for `StaticGroup` objects."""
+
+    content_type = DynamicModelChoiceField(
+        queryset=ContentType.objects.all(),
+        query_params={"feature": "static_groups"},
+    )
+    tenant = DynamicModelChoiceField(queryset=Tenant.objects.all(), required=False)
+
+    class Meta:
+        model = StaticGroup
+        fields = (
+            "name",
+            "content_type",
+            "description",
+            "tenant",
+            "tags",
+        )
+
+
+class StaticGroupFilterForm(NautobotFilterForm):
+    model = StaticGroup
+    q = forms.CharField(required=False, label="Search")
+    content_type = CSVContentTypeField(
+        queryset=ContentType.objects.filter(FeatureQuery("static_groups").get_query()).order_by("app_label", "model"),
+        required=False,
+    )
+    tenant = DynamicModelChoiceField(queryset=Tenant.objects.all(), required=False)
+    tags = TagFilterField(model)
+
+
+class StaticGroupBulkEditForm(NautobotBulkEditForm):
+    pk = forms.ModelMultipleChoiceField(
+        queryset=StaticGroup.objects.all(),
+        widget=forms.MultipleHiddenInput(),
+    )
+    description = forms.CharField(max_length=CHARFIELD_MAX_LENGTH, required=False)
+    tenant = DynamicModelChoiceField(queryset=Tenant.objects.all(), required=False)
+
+    class Meta:
+        nullable_fields = ["tenant"]
+
+
+class StaticGroupBulkAssignForm(BootstrapMixin, BulkEditForm):
+    content_type = forms.ModelChoiceField(
+        queryset=ContentType.objects.filter(FeatureQuery("static_groups").get_query()).order_by("app_label", "model"),
+        widget=forms.HiddenInput(),
+    )
+    create_and_assign_to_new_group_name = forms.CharField(
+        required=False,
+        label="Add to new group",
+        help_text="Create a new group with this name and assign the selected objects to it.",
+    )
+
+    def __init__(self, model, *args, **kwargs):
+        super().__init__(model, *args, **kwargs)
+        self.fields["content_type"].initial = ContentType.objects.get_for_model(model)
+        self.fields["pk"] = forms.ModelMultipleChoiceField(
+            queryset=model.objects.all(),
+            widget=forms.MultipleHiddenInput(),
+            required=False,
+        )
+        self.fields["add_to_groups"] = DynamicModelMultipleChoiceField(
+            queryset=StaticGroup.objects.all(),
+            required=False,
+            query_params={"content_type": model._meta.label_lower},
+            label="Add to existing group(s)",
+        )
+        self.fields["remove_from_groups"] = DynamicModelMultipleChoiceField(
+            queryset=StaticGroup.objects.all(),
+            required=False,
+            query_params={"content_type": model._meta.label_lower},
+            label="Remove from group(s)",
+        )
```

Comment:
These two fields input value should be mutually exclusive

Reply:
Easier to add as form backend logic than as UI frontent logic, definitely.